query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/PostgresFunctionConverter.java

```diff
@@ -0,0 +1,280 @@
+package org.hypertrace.core.query.service.postgres.converters;
+
+import static org.hypertrace.core.query.service.QueryFunctionConstants.DATA_TIME_CONVERT;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_AVGRATE;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_CONCAT;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_CONDITIONAL;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_COUNT;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_DISTINCTCOUNT;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_PERCENTILE;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Function;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.api.ValueType;
+import org.hypertrace.core.query.service.postgres.TableDefinition;
+
+public class PostgresFunctionConverter {
+  private static final String POSTGRES_CONCAT_FUNCTION = "CONCATSKIPNULL";
+  private static final String DEFAULT_AVG_RATE_SIZE = "PT1S";
+  private static final String SUM_FUNCTION = "SUM";
+
+  private final TableDefinition tableDefinition;
+  private final PostgresFunctionConverterConfig config;
+
+  public PostgresFunctionConverter(
+      TableDefinition tableDefinition, PostgresFunctionConverterConfig config) {
+    this.tableDefinition = tableDefinition;
+    this.config = config;
+  }
+
+  public PostgresFunctionConverter(TableDefinition tableDefinition) {
+    this(tableDefinition, new PostgresFunctionConverterConfig());
+  }
+
+  public String convert(
+      ExecutionContext executionContext,
+      Function function,
+      java.util.function.Function<Expression, String> argumentConverter) {
+    switch (function.getFunctionName().toUpperCase()) {
+      case QUERY_FUNCTION_COUNT:
+        return this.convertCount();
+      case QUERY_FUNCTION_PERCENTILE:
+        // Computing PERCENTILE in Postgres is resource intensive. T-Digest calculation is much
+        // faster
+        // and reasonably accurate, so support selecting the implementation to use
+        return this.toPostgresPercentile(function, argumentConverter);
```

Comment:
nit: again, I don't think we need to prefix only a few functions explictly`toPostgresPercentile` when we are already dealing with PostgresFunctionConverter.

would go with `toPercentile, toConcat, toDistinctCount, toCount` etc.

Reply:
Can you do this as follow up?